src/dataset-builder/CohortEditor.ts

```diff
@@ -206,75 +229,105 @@ type AddCriteriaSelectorProps = {
   index: number;
   criteriaGroup: CriteriaGroup;
   updateCohort: Updater<Cohort>;
-  snapshotBuilderSettings: SnapshotBuilderSettings;
+  dataset: DatasetModel;
   onStateChange: OnStateChangeHandler;
   cohort: Cohort;
 };
 
 const AddCriteriaSelector: React.FC<AddCriteriaSelectorProps> = (props) => {
-  const { index, criteriaGroup, updateCohort, snapshotBuilderSettings, onStateChange, cohort } = props;
-  return h(GroupedSelect<CriteriaOption>, {
-    styles: { container: (provided) => ({ ...provided, width: '230px', marginTop: wideMargin }) },
-    isClearable: false,
-    isSearchable: false,
-    options: [
-      {
-        label: 'Domains',
-        options: _.map(
-          (domainOption) => ({
-            value: addKindToDomainOption(domainOption),
-            label: domainOption.category,
-          }),
-          snapshotBuilderSettings.domainOptions
-        ),
-      },
-      {
-        label: 'Program Data',
-        options: _.map((programDataOption) => {
-          return {
-            value: programDataOption,
-            label: programDataOption.name,
-          };
-        }, snapshotBuilderSettings.programDataOptions),
-      },
-    ],
-    'aria-label': addCriteriaText,
-    placeholder: addCriteriaText,
-    value: null,
-    onChange: (x) => {
-      if (x !== null) {
-        if (x.value.kind === 'domain') {
-          onStateChange(domainCriteriaSelectorState.new(cohort, criteriaGroup, x.value));
-        } else {
-          const criteria = criteriaFromOption(x.value);
-          if (criteria !== undefined) {
-            updateCohort(_.set(`criteriaGroups.${index}.criteria.${criteriaGroup.criteria.length}`, criteria));
+  const {
+    index,
+    criteriaGroup,
+    updateCohort,
+    dataset: { id: datasetId, snapshotBuilderSettings },
+    onStateChange,
+    cohort,
+  } = props;
+  return (
+    snapshotBuilderSettings &&
+    h(GroupedSelect<CriteriaOption>, {
+      styles: { container: (provided) => ({ ...provided, width: '230px', marginTop: wideMargin }) },
+      isClearable: false,
+      isSearchable: false,
+      options: [
+        {
+          label: 'Domains',
+          options: _.map(
+            (domainOption) => ({
+              value: addKindToDomainOption(domainOption),
+              label: domainOption.category,
+            }),
+            snapshotBuilderSettings.domainOptions
+          ),
+        },
+        {
+          label: 'Program Data',
+          options: _.map((programDataOption) => {
+            return {
+              value: programDataOption,
+              label: programDataOption.name,
+            };
+          }, snapshotBuilderSettings.programDataOptions),
+        },
+      ],
+      'aria-label': addCriteriaText,
+      placeholder: addCriteriaText,
+      value: null,
+      onChange: async (x) => {
```

Comment:
is there a better name for x?

Reply:
Yes, it's the selected option type, I can rename this.